src/sysext/sysext.c

```diff
@@ -349,19 +400,36 @@ static int unmerge_hierarchy(
                 }
 
                 log_info("Unmerged '%s'.", p);
+                n_unmerged++;
         }
 
-        return 0;
+        return n_unmerged;
 }
 
 static int unmerge_subprocess(
                 ImageClass image_class,
-                char **hierarchies) {
+                char **hierarchies,
+                const char *workspace) {
 
         int r, ret = 0;
 
+        /* Mark the whole of /run as MS_SLAVE, so that we can mount stuff below it that doesn't show up on
+         * the host otherwise. */
+        r = mount_nofollow_verbose(LOG_ERR, NULL, "/run", NULL, MS_SLAVE|MS_REC, NULL);
+        if (r < 0)
+                return log_error_errno(r, "Failed to remount /run/ MS_SLAVE: %m");
```

Comment:
drop the log msg, just do `return r;` directly. After all you are already using `mount_nofollow_verbose()` above, which means you already get log msgs about all errors.

Reply:
it is?